src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,684 @@
+                            "No provider can be found that supports the "
+                            + "specified algorithm and parameters");
+                    if (!skipDebug && pdebug != null) {
+                        pdebug.println(
+                            "obj was not an instance of KDFSpi (should not "
+                            + "happen)");
+                    }
+                    // continue to next iteration
+                } else if (t.hasNext()) {
+                    return new KDF(new Delegate(spiObj, s.getProvider()), t,
+                                   algorithm, kdfParameters);
+                } else { // no other choices, lock down provider
+                    return new KDF(new Delegate(spiObj, s.getProvider()),
+                                   algorithm, kdfParameters);
+                }
+            } catch (NoSuchAlgorithmException e) {
+                lastException =
+                        new InvalidAlgorithmParameterException(
+                            "No provider can be found that supports the "
+                            + "specified algorithm and parameters");
+                if (!skipDebug && pdebug != null) {
+                    pdebug.println(e.toString());
+                }
+                // continue to next iteration
+            }
+        }
+        if (lastException != null) {
+            throw lastException;
+        } else {
+            throw new NoSuchAlgorithmException(
+                "Algorithm " + algorithm + " not available");
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} object that implements the specified algorithm from
+     * the specified provider and is initialized with the specified parameters.
+     * The specified provider must be registered in the security provider list.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use.
+     *     See the {@code KDF} section in the <a href=
+     *     "{@docRoot}/../specs/security/standard-names.html#kdf-algorithms">
+     *     Java Security Standard Algorithm Names Specification</a>
+     *     for information about standard KDF algorithm names.
+     * @param kdfParameters
+     *     the {@code KDFParameters} used to configure the derivation
+     *     algorithm or {@code null} if no parameters are provided
+     * @param provider
+     *     the provider to use for this key derivation
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if the specified provider does not support the specified {@code KDF}
+     *     algorithm
+     * @throws NoSuchProviderException
+     *     if the specified provider is not registered in the security provider
+     *     list
+     * @throws InvalidAlgorithmParameterException
+     *     if the specified provider does not support a {@code KDFSpi}
+     *     implementation for the specified algorithm and parameters
```

Comment:
Do not mention `algorithm` in this `@throws`. It's already covered by `@throws NSAE`. Same below.

Reply:
It that reviewer was me, I'm sorry I changed my mind. The current sentence is a little misleading IMO.